### ISSUE TITLE
Default XMPP priority

### ISSUE THREAD

**Issue by mccreary** [ChatSecure/ChatSecure-iOS#68]
Can you raise the default XMPP priority?  It is currently 0, and Google uses 24 for their clients.  I want the ChatSecure session to have higher priority, so it will preempt any non-OTR-capable sessions that might be left running.  How about using 30?

Alternatively, a way to set the priority in the app settings would be a useful enhancement.

EDIT:  Both Google's Talk app for Android and the Gmail web client use a priority of 24, although the Gmail web client drops the priority to 0 when away.

**Comment by jrasanen** [comment]
Anything new on this? Priority is quite essential if you want to receive same XMPP messages to multiple devices. I.e. Phone + computer.

**Comment by juk13** [comment]
Anything new on this? Please Implement it. I need it.

**Comment by RinoKRD** [comment]
T

**Comment by peteeckel** [comment]
+1 from me. 

The problem is that priority 0 means that if there is any other XMPP client connected to the account, ChatSecure won't receive the message **ever**.

Use case: Usually I am reading my XMPP messages at one of my computers. All of them are set up so that the priority is below 50 when offline/unavailable, above 50 when available, so the messages get routed to a machine I am likely to be currently using at any time.

The issue arises when I am **not** at any of my computers - in that case (and only in that case) I want them to be routed to my phone. But since the default priority of 0 is the lowest possible one, there is no way I can configure my other clients so that this goal is achieved - 0 results in messages only getting routed to the phone if no other client is **connected** at all. If I forget to (or rather, choose not to) close all clients on all computers I don't get anything at ChatSecure. 

Setting the computers offline priority to a negative value is currently the only workaround I see, but that's not too good either as no messages will get routed to any of the computers at all, which breaks sending OTR encrypted messages. 

So please consider implementing setting the XMPP priority, or at least raise the default priority to a value higher than 0.

**Comment by chrisballinger** [comment]
@peteeckel Does your server have MAM enabled?

**Comment by peteeckel** [comment]
Now it does :-)

I'll continue my tests as soon as possible, though having the priority configurable would still be the more flexible solution.

**Comment by chrisballinger** [comment]
Aye, priority should be easy enough to add but it's sort of antiquated and
I don't want to complicate things by adding a ton of options. We don't
support MAM yet actually, but once it's added the whole priority thing will
be less of an issue.

**Comment by peteeckel** [comment]
It will help some, but XEP-0313 is still experimental, and for MAM to be effective AFAIK all clients that are used simultaneously have to support it, which might take a while. Configurable priorities on the other hand are quite commonplace across most clients I know.

In addition to that I'm afraid it won't work with OTR encryption at all, as both clients, sender and recipient, need to be online at the same time so they can negotiate a session key. OMEMO should be fine, but isn't supported i.e. by Adium.

**Comment by chrisballinger** [comment]
Yeah, agreed. Btw... Adium is unsafe to use and is no longer maintained.

**Comment by mcnesium** [comment]
So why not just setting the default priority to 30 as OP suggested? People who use Chatsecure only will not be affected at all, and any other user who has multiple clients can then update the priority settings of their other clients according to their needs.
